src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/fs/TeeStepTest.java

```diff
@@ -76,6 +76,47 @@ public void smokes() throws Exception {
         });
     }
 
+    @Test
+    public void create() throws Exception {
+        rr.then(r -> {
+            r.createSlave("remote", null, null);
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(
+                    "node('remote') {\n" +
+                            "  tee('x.log') {\n" +
+                            "    echo 'first message'\n" +
+                            "  }\n" +
+                            "  tee('x.log') {\n" +
+                            "    echo 'second message'\n" +
+                            "  }\n" +
+                            "  echo(/got: ${readFile('x.log').trim().replaceAll('\\\\s+', ' ')}/)\n" +
+                            "}", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            r.waitForCompletion(b);
+            r.assertLogContains("got: second message", b);
+        });
+    }
+
+    @Test
```

Comment:
This does sound like a test for a fix of JENKINS-54346—of which JENKINS-55505 sounds like a duplicate, though I do not see a test for that specifically (two `bat` statements inside one `tee`).